File: android/app/src/main/java/com/texasgamer/zephyr/util/resource/ResourceProvider.java
package com.texasgamer.zephyr.util.resource;

import android.content.Context;

import androidx.annotation.NonNull;
import androidx.annotation.StringRes;

/**
 * Implementation of {@link IResourceProvider} which uses {@link Context} to resolve resources.
 */
public class ResourceProvider implements IResourceProvider {

    private Context mContext;

    public ResourceProvider(@NonNull Context context) {
        mContext = context;
    }

    @NonNull
    @Override
    public String getString(@StringRes int stringRes) {
        return mContext.getString(stringRes);
    }

    @NonNull
    @Override
    public String getString(@StringRes int stringRes, Object... formatArgs) {
        return mContext.getString(stringRes, formatArgs);
    }
}

File: android/app/src/main/java/com/texasgamer/zephyr/util/resource/ResourceProvider.kt
package com.texasgamer.zephyr.util.resource

import android.content.Context
import android.graphics.drawable.Drawable
import androidx.annotation.DrawableRes
import androidx.annotation.StringRes
import androidx.core.content.ContextCompat

/**
 * Implementation of [IResourceProvider] which uses [Context] to resolve resources.
 */
class ResourceProvider(private val mContext: Context) : IResourceProvider {
    override fun getString(@StringRes stringRes: Int): String {
        return mContext.getString(stringRes)
    }

    override fun getString(@StringRes stringRes: Int, vararg formatArgs: Any): String {
        return mContext.getString(stringRes, *formatArgs)
    }

    override fun getDrawable(@DrawableRes drawableRes: Int): Drawable? {
        return ContextCompat.getDrawable(mContext, drawableRes)
    }
}